sdk/eventhub/azure-eventhub/azure/eventhub/_producer.py

```diff
@@ -206,8 +203,6 @@ def _wrap_eventdata(
                     raise ValueError(
                         "The partition_key does not match the one of the EventDataBatch"
                     )
-                for message in event_data.message._body_gen:  # pylint: disable=protected-access
```

Comment:
need `trace_message` for message in the _body_gen?
otherwise it looks like we're not handling OT in this case.